enso/experiment/doc_rep.py

```diff
@@ -0,0 +1,88 @@
+from finetune import SequenceLabeler
+from finetune.base_models import RoBERTa
+
+from enso.registry import Registry, ModeKeys
+from enso.experiment.finetuning import FinetuneSequenceLabel
+
+
+@Registry.register_experiment(ModeKeys.SEQUENCE, requirements=[("Featurizer", "TextContextFeaturizer")])
+class SidekickSeqLab(FinetuneSequenceLabel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_config.update(
+            dict(
+                # general params that differ from finetune
+                base_model=RoBERTa,
+                batch_size=4,
+                predict_batch_size=10,
+                val_size=0.0,
+                crf_sequence_labeling=False,
+                low_memory_mode=True,
+                class_weights="log",
+                # auxiliary-specific params
+                use_auxiliary_info=True,
+                context_dim=4,
+                default_context={
+                    'left': 0,
+                    'right': 0,
+                    'top': 0,
+                    'bottom': 0,
+                },
+                n_context_embed_per_channel=48,
+                context_in_base_model=True,
+                n_layers_with_aux=-1)
+        )
+        self.model_config.update(kwargs)
+        self.model = SequenceLabeler(**self.model_config)
+
+    def fit(self, X, y):
+        text, context = zip(*X)
+        self.model.fit(text, y, context=context)
+
+    def predict(self, X, **kwargs):
+        text, context = zip(*X)
+        return self.model.predict(text, context=context)
+
+@Registry.register_experiment(ModeKeys.SEQUENCE, requirements=[("Featurizer", "TextContextFeaturizer")])
+class RoBERTaSeqLab(SidekickSeqLab):
```

Comment:
Maybe inherit from FinetuneSequenceLabeler again -- or if you do want to inherit from SidekickSeqLab you can delete fit + predict because the superclass already behaves the same way.

Reply:
Maybe best option would be to have everything inherit from this class and just override the inits on the other classes.

Reply:
that's what I originally did but the param values are quite tricky so I didn't trust myself to get them right. I chose to do it this way to we're matched one-to-one with the code in doc_rep, way easier to ensure consistency

Reply:
Sorry I agree that makes sense, I just meant the fit() and predict() calls are constant across all classes so you don't need to respecify them .